mousebender/simple.py

```diff
@@ -183,14 +200,54 @@ def handle_starttag(self, tag, attrs_list):
         # Links in the simple repository MAY have a data-yanked attribute which
         # may have no value, or may have an arbitrary string as a value.
         yanked = "data-yanked" in attrs, attrs.get("data-yanked") or ""
-
-        self.archive_links.append(
+        self.update_index(
             ArchiveLink(filename, url, requires_python, hash_, gpg_sig, yanked)
         )
 
+    def update_index(self, archive: ArchiveLink):
+        # old way
+        self.archive_links.append(archive)
+
+        # new way, indexed values
+
+        # Dict[str, Dict[str, Tuple[Optional[ArchiveLink], Optional[Dict[Tag, ArchiveLink]]]]]
+        try:
+            proj, ver, build_tag, tags = packaging.utils.parse_wheel_filename(
+                archive.filename
+            )
+
+            if proj not in self.index:
+                self.index[proj] = {}
+            if ver not in self.index[proj]:
+                self.index[proj][ver] = (None, {tags: archive})
+            else:
+                # handle immutability of a Tuple by re-creating it, pull out the old members:
+                sdist_lnk = self.index[proj][ver][0]
+                archive_link_dict = self.index[proj][ver][1] or {}
+                # add the new info:
+                archive_link_dict[tags] = archive
+                # and re-create it:
+                self.index[proj][ver] = (sdist_lnk, archive_link_dict)
+
+        except packaging.utils.InvalidWheelFilename as bad_file_name:
+            # This is either an sdist or something else.
+
+            try:
+                proj, ver = packaging.utils.parse_sdist_filename(archive.filename)
+                if proj not in self.index:
+                    self.index[proj] = {}
+                if ver not in self.index[proj]:
+                    self.index[proj][ver] = (archive, {})
+
+            except packaging.utils.InvalidSdistFilename as bad_file_name:
+                pass
+                # warnings.warn(
```

Comment:
Not sure if we want to relay 'rejected' information here? I might be inclined to include the `logging` module to send information messages to the user if they are so inclined. Thoughts?

Reply:
Probably just drop them as they cannot be installed by pip.